fixieai/agents-ts/src/serve.ts

```diff
@@ -214,34 +221,40 @@ export default async function serve({
         return;
       }
 
-      const body = req.body as AgentResponse;
+      const body = req.body as SerializedMessageEnvelope;
+      const reqMessage = await messageOfSerializedMessage(body.message);
+
+      function serializedMessageOfMessage(message: Message): SerializedMessage {
+        const result: Partial<SerializedMessage> = _.pick(message, 'text');
+        result.embeds = _.mapValues(message.embeds, (e) => e.serialize());
+        return result as SerializedMessage;
+      }
+
+      async function messageOfSerializedMessage(serializedMessage: SerializedMessage): Promise<Message> {
+        const result: Partial<Message> = _.pick(serializedMessage, 'text');
+        const embedPromises = Object.entries(serializedMessage.embeds ?? {}).map(async (
+          [embedName, embed],
+        ) => [embedName, await Embed.fromUri(embed.content_type, embed.uri)]);
```

Comment:
IMO we should let the agent code initiate this if they want the data directly. (But they may not -- for example they may want to pass that URI down to another service or w/e)